### ISSUE TITLE
Something wrong with 'Allow Script Tags in Markdown'

### ISSUE THREAD

**Issue by GreenEyedAndy** [RickStrahl/MarkdownMonster#903]
If the menuitem is checked and I have mermaid in my markdown the mermaid is not rendered.
If I now uncheck the menuitem I get the warning that mermaid etc. can't be rendered, but now the mermaid-diagram is visible.
If I now check to state the menuitem I see the menuitem is checked (I unchecked it in the step before).
Something must be wrong with the binding.

**Comment by RickStrahl** [comment]
What happens if you restart MM? 

I think the value is being set as a restart works (for me). But as you say it doesn't apply immediately, which is it should on any re-render (which requires a full re-load which may require closing a tab and re-opening - but even that doesn't seem to fix the state issue at the moment - a restart does).

I have to take a look. A bit of work was done on Mermaid for 2.3.2 so it's possible something was broken there that doesn't respect the flag.

Incidentally I'm thinking that Mermaid may not require any special consideration for script in a page, because Mermaid doesn't require script inside of the Markdown - only in the header  injected by MM internally and which can't be changed by user code. So theoretically MM could allow Mermaid without any script attack issues.

**Comment by RickStrahl** [comment]
So I took a look and I think the problem was more of a logic error that wasn't explained well :smile:

When you turn off the Script Rendering MM also explicitly disabled Mermaid functionality (disabled the switch). Which will make it not work. Disabling/Enabling Mermaid currently requires a restart as the RenderExtensions are loaded on MM startup.

So... after some review there's really no reason to require script rendering to be enabled for Mermaid because it doesn't inject any script code into the markdown itself. It only adds it into the header. So now I no longer disable Mermaid when the scripting is disabled and Mermaid will continue to work without scripting enabled - even after a restart.

In short, Mermaid now always works as long as the Mermaid configuration option is set.

**Comment by RickStrahl** [comment]
These updates are in 2.3.4.
